altair/utils/data.py

```diff
@@ -49,7 +51,10 @@ def limit_rows(data, max_rows=5000):
 
     If max_rows is None, then do not perform any check.
     """
-    check_data_type(data)
+    try:
+        check_data_type(data)
+    except TypeError:
+        return data
```

Comment:
I think it would be cleaner to get rid of the ``check_data_type`` function, because it doesn't really help now that more data types are supported. For example, here the type checks are already accomplished below... all that's needed is an ``else`` statement that returns the data.

Reply:
Sounds good, updated.